### PR TITLE
HBHE M2 = HPD old code cleanup

### PR DESCRIPTION
**Pull request by mariadalfonso**
Given that we are switching the legacy re-reco to the "Phase1"code with the #17676.
I also removed the various if statement for the HPD-QIE8.

Now take the noise from the database instead of a flat numbers from the python code
the ADCquantization noise  use the exact definition instead of a parametrized version

expect minor changes in the noise hits

### CODE REVIEW

**Comment by slava77** [comment]
@mariadalfonso 
if I understand correctly, this PR changes behavior and needs #17676 to be merged to proceed with tests.
Please confirm.
Thank you.

**Comment by abdoulline** [comment]
90X #17676  -> now  91X  #17688 for this PR

**Comment by mariadalfonso** [comment]
@slava77 
To see the effect of this PR on some of the workflow we do not need to wait for the #17676 ; but this PR should be merged only if the  #17676, that is when the completely dismiss the "run1/run2-2016" code in favor of the "Phase1" for all the era.

**Comment by slava77** [comment]
@cmsbuild please test
.. to be re-tested after  #17676

**Comment by mariadalfonso** [comment]
@slava77 
What I do with this PR, should I forward port to the 91X (as the 17676 ->17688) ?

**Comment by slava77** [comment]
@mariadalfonso this is already a 91X PR.
I didn't get to #17688 yet.
Once it's signed off, this PR will be reviewed.

**Comment by slava77** [comment]
@cmsbuild please test

the last set of tests has expired

File: RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc [hunk 1]
```diff
@@ -122,15 +122,15 @@ namespace FitterFuncs{
 	//Interpolate the fit (Quickly)
 	funcHPDShape(pulse_shape_, pars[0],pars[1],psFit_slew[time]);
 	for (j=0; j<nbins; ++j) {
-	  psFit_erry2[j]  = psFit_erry[j]*psFit_erry[j] + pulse_shape_[j]*pulse_shape_[j]*pulseJitter_;
+	  psFit_erry2[j]  = psFit_erry2[j] + pulse_shape_[j]*pulse_shape_[j]*pulseJitter_;
```
Comment (slava77) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: note to self: addPulseJitter_ current default is false and this scope is not running

**Comment by slava77** [comment]
[the following plots were made with CMSSW_9_1_X_2017-03-09-1100 + #17688 as a baseline and 
"CMSSW_9_1_X_2017-03-09-1100 + #17688 + this PR" as "new"]

In MC workflows
hbhe reco chi2 changes more in the barrel and almost unchanged in endcap
Here is an example plot from wf 10071  flat-pt QCD dijet sample
all_sign882vssign874_qcd13tevflatpt15s3000in2017wf10071p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_chi210
all_sign882vssign874_qcd13tevflatpt15s3000in2017wf10071p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_chi220

this is matched with an increase in 3-pulse fits
all_sign882vssign874_qcd13tevflatpt15s3000in2017wf10071p0c_log2maxhbherechitssorted_hbhereco__reco_obj_obj_flags 0_58

The change is more significant in HE in presence of PU, visible both in data and MC
e.g. 2016E wf 136.761
HB
all_sign882vssign874_runjetht2016ewf136p761c_log10hbherechitssorted_hbhereco__rereco_obj_obj_chi210
vs HE
all_sign882vssign874_runjetht2016ewf136p761c_log10hbherechitssorted_hbhereco__rereco_obj_obj_chi220

There are fairly small variations in the hit energy distributions with about 2-3% increase in the number of hits with E>1MeV in HE in events with PU, similar in data and MC:
in 2016E data:
all_sign882vssign874_runjetht2016ewf136p761c_log10hbherechitssorted_hbhereco__rereco_obj_obj_energy13

In HB data and MC appear to go in opposite directions:
2016E data JetHT
all_sign882vssign874_runjetht2016ewf136p761c_log10hbherechitssorted_hbhereco__rereco_obj_obj_energy3
vs 25202 ttbar PU35 (similar PU as in 2016E)
all_sign882vssign874_ttbarpuwf25202p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_energy3

These changes in hit energy do not contribute significantly to the PF cluster energy distributions or to the calo and PF jet response and are not visible in MET either.

As expected, the 2018 workflow has no changes in the endcap (this PR affects only HPD channels).

On the technical side, the hbheprereco time is up by about 8%, and the total reco time is up by around 1%.

Overall, the behavior appears to be roughly as expected and changes are small enough to not affect whatever lower level calibrations are to be done with 90X. So, I think that this can go to 91X.

**Comment by slava77** [comment]
@mariadalfonso please check and confirm the chi2 behavior HB vs HE is expected.
Thank you.

**Comment by slava77** [comment]
@mariadalfonso 
please respond to https://github.com/cms-sw/cmssw/pull/17680#issuecomment-287581470
thank you

**Comment by kpedro88** [comment]
@slava77 should we retest now that #17688 is merged?

**Comment by slava77** [comment]
On 3/20/17 9:07 AM, Kevin Pedro wrote:
> @slava77 <https://github.com/slava77> should we retest now that #17688
> <https://github.com/cms-sw/cmssw/pull/17688> is merged?
> 

I was waiting for baseline to show up first.
It's still not there yet.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@kpedro88 @abdoulline 
maybe you can comment on the difference in chi2 between HB and HE
is digitization/quantization scale different or something else similar in effect on the pulse shape

Looking vs energy, it doesn't appear as if the effect is noise related
wf10071_chi2_vs_e_hb
wf10071_chi2_vs_e_he

File: RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc (hunk 1 above)
Comment (mariadalfonso) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: @slava77 , yes correct the addPulseJitter_ is set to off , but worth to correct those bugs anyway

**Comment by mariadalfonso** [comment]
@slava77 
This PR takes 
1) ADC quantization from exact definitions, some test can be found in #17037
2) pedestal width from the DB #17129
From plots in these PR, I do not expect changes in the chi2

3) change also the gaussian constraint on the pedestal

**Comment by abdoulline** [comment]
@slava77

the difference I see between HB and HE (e.g. for the same end RecHit 
energy) comes from different sampling (scint./absorber ratio)

(1) pfotostatistics. Factor ~1.8 more light (p.e.) in HB for the same 
"end" RecHit energy. So, photostatistics error is bigger in HE [*]

(2) similarly ADC quantization scale is lower in HE by factor ~1.8.
less fC = lower ADC scale for the same "end" RecHit energy wrt HB - 
means bigger relative quantization error in HE.

Well, it might mean that if something is changed, Chi2 effect should be 
less pronounced in HE (with its bigger error terms) ?

    Salavat

[*]  simHitToPhotoelectrons is similar for HB and HE
https://github.com/cms-sw/cmssw/blob/CMSSW_9_0_X/SimCalorimetry/HcalSimAlgos/src/HcalSimParameters.cc#L56
(samplingFactor ~1.8 difference compenasated by ~1.8 fCtoGeV)

Means Poisson smearing is more "crude" for HE 
as ~same "npe" generated here for the same SimHit energy in HB and HE
https://github.com/cms-sw/cmssw/blob/CMSSW_9_0_X/SimCalorimetry/CaloSimAlgos/src/CaloHitResponse.cc#L166

but in HE SimHit energy is ~1.8 smaller than in HB for the same resulting 
RecHit.

**Comment by mariadalfonso** [comment]
@abdoulline 

> (1) pfotostatistics. Factor ~1.8 more light (p.e.) in HB for the same
> "end" RecHit energy. So, relative photostatistics error is bigger in HE [*]

This PhotoStat term didn't change in this PR, as was already active in the release for both the HB and HE and HEP17

> 
> (2) similarly ADC quantization scale is lower in HE by factor ~1.8.
> less fC = lower ADC scale for the same "end" RecHit energy wrt HB -
> means bigger relative quantization error in HE.
> 
> Well, it might mean that if something is changed, Chi2 effect should be
> less pronounced in HE (with its bigger error terms) ?
> 
> S.

**Comment by abdoulline** [comment]
Yes, I know that photostat. term didn't change in this PR, Maria. 
I was just answering Slava's question about a difference in general between 
HE and HB.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17680	9eb66e9
- code changes are as described
- jenkins tests pass and comparisons showed differences only in phase-1 workflows in the state before #17688 is merged; most recent comparisons show differences in other workflows as well [consistent with observations in local tests]
- local tests on top of #17688 show somewhat expected behavior

https://github.com/cms-sw/cmssw/pull/17680#issuecomment-287580437 : 
changes are small enough to not affect whatever lower level calibrations are to be done with 90X. So, I think that this can go to 91X.

**Comment by davidlange6** [comment]
+1

**Pull request merged**